### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by rrong**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+
+from setuptools import setup
+
+setup(name='tap-bing-hotel-ads',
+      version="1.0.0",
+      description='Singer.io tap for extracting data from the Bing hotel ads api',
```
Comment (nehiljain) [inline setup.py]: nit: add bing hotel ads api link

**Review by nehiljain — COMMENTED** [review]
Let's add tests

File: tap_bing_hotel_ads/__init__.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        token = new_token
+        CONFIG['oauth_access_token'] = token['access_token']
+        CONFIG['oauth_refresh_token'] = token['refresh_token']
+
+    session = OAuth2Session(client_id, token=token, auto_refresh_url=refresh_url, auto_refresh_kwargs=extra, token_updater=token_saver)
+    return session
+
+async def poll_report(customer_id, account_id, job_id, start_date, end_date):
+    download_url = None
+    poll_url = "https://partner.api.bingads.microsoft.com/Travel/v1/Customers({})/Accounts({})/ReportJobs('{}')".format(customer_id, account_id, job_id)
```
Comment (nehiljain) [inline tap_bing_hotel_ads/__init__.py]: should we add linter to this repo? This line is very long and can be easily split into multiple lines

File: tap_bing_hotel_ads/__init__.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+                i,
+                MAX_NUM_REPORT_POLLS,
+                job_id,
+                start_date,
+                end_date))
+            response = SESSION.get(poll_url)
+            resp_obj = response.json()
+            if resp_obj["Status"] == 'Completed':
+                if resp_obj["Url"]:
+                    download_url = resp_obj["Url"]
```
Comment (stevenli1141) [inline tap_bing_hotel_ads/__init__.py]: nit: single/double quotes should be consistent

File: tap_bing_hotel_ads/__init__.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+                break
+
+            if i == MAX_NUM_REPORT_POLLS:
+                LOGGER.info('Generating report timed out: {} - from {} to {}'.format(
+                        job_id,
+                        start_date,
+                        end_date))
+                return False, ''
+            else:
+                await asyncio.sleep(REPORT_POLL_SLEEP)
```
Comment (nimwijetunga) [inline tap_bing_hotel_ads/__init__.py]: Did you test this? I think you should be using an async with 
`async with metrics.job_timer('generate_report'):`

Comment (rrong) [reply tap_bing_hotel_ads/__init__.py]: I think this is fine. It looks like async with just runs the resource acquisition in a different context?

https://docs.python.org/3/reference/compound_stmts.html#the-async-with-statement

**Review by nehiljain — APPROVED** [review]

**Pull request merged**